pkg/operator/ceph/pool/controller.go

```diff
@@ -403,10 +404,20 @@ func configureRBDStats(clusterContext *clusterd.Context, clusterInfo *cephclient
 	return nil
 }
 
-func (r *ReconcileCephBlockPool) cancelMirrorMonitoring(cephBlockPoolName string) {
-	// Cancel the context to stop the go routine
-	r.blockPoolContexts[cephBlockPoolName].internalCancel()
+func blockPoolChannelKeyName(cephBlockPool *cephv1.CephBlockPool) string {
+	return fmt.Sprintf("%s-%s", cephBlockPool.Namespace, cephBlockPool.Name)
```

Comment:
Shall we use a NamespacedName.String() similar to the object keyname?

Reply:
I was considering that. This merely keeps exactly what existed from before. I had an initial concern about the update scenario, but since the operator is restarted for the update, I don't think there will be any pre-existing channels to conflict with. Does that sound correct to you also?

Reply:
Agreed, the key is just used in-memory so the upgrade should be fine